api/screenshot/model.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package screenshot
+
+import (
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	mapset "github.com/deckarep/golang-set"
+
+	"github.com/web-platform-tests/wpt.fyi/shared"
+)
+
+// MaxItemsInResponse is the maximum number of hashes that will be returned
+// when asked for recent screenshots.
+const MaxItemsInResponse = 10000
+
+var (
+	// ErrInvalidHash is the error when a hash string is invalid.
+	ErrInvalidHash = errors.New("invalid hash string")
+	// ErrUnsupportedHashMethod is the error when the requested hash method
+	// is not supported.
+	ErrUnsupportedHashMethod = errors.New("hash method unsupported")
+)
+
+// Screenshot is the entity stored in Datastore for a known screenshot hash and
+// its metadata.
+type Screenshot struct {
+	HashDigest string
+	HashMethod string
+	Labels     []string
+	// These two fields can be left empty and will be filled by Store().
+	Counter  int
+	LastUsed time.Time
+}
+
+// NewScreenshot creates a new Screenshot with the given labels (empty labels
+// are omitted).
+func NewScreenshot(labels []string) *Screenshot {
```

Comment:
IMO, callers shouldn't have to wrap labels in an array when there's only one parameter type.

```suggestion
func NewScreenshot(labels ...string) *Screenshot {
```

Reply:
Yeah it used to have more parameters. Fixed.